ome-model: New package to replace ome-xml (bioformats)

rleigh-codelibre opened the pull request:
Switch to using ome-model (ome-model.git) in place of ome-xml (bioformats.git).  Some references to ome-xml remain with respect to the documentation, because the library and doxygen documentation are still named "ome-xml".  We could rename it as a followup task.

The scc merge job and build jobs have been updated to merge and clone ome-model in place of bioformats.

Commit:
ome-model: New package to replace ome-xml (bioformats)

Commit:
Clean up remaining ome-xml references

sbesson commented on docs/sphinx/building.rst:
```diff
@@ -478,7 +478,7 @@ out.  Run::
     mkdir build
     cd build
-    cmake -G Ninja -Dome-common-dir=/path/to/ome-common-cpp -Dome-xml-dir=/path/to/bioformats -Dome-files-dir=/path/to/ome-files -Dome-qtwidgets-dir=/path/to/ome-qtwidgets -Dbuild-prerequisites=OFF -Dome-superbuild_BUILD_gtest=ON -Dbuild-packages=ome-qtwidgets -DCMAKE_INSTALL_PREFIX=/install/dir /path/to/superbuild/git/repo
+    cmake -G Ninja -Dome-common-dir=/path/to/ome-common-cpp -Dome-model-dir=/path/to/bioformats -Dome-files-dir=/path/to/ome-files -Dome-qtwidgets-dir=/path/to/ome-qtwidgets -Dbuild-prerequisites=OFF -Dome-superbuild_BUILD_gtest=ON -Dbuild-packages=ome-qtwidgets -DCMAKE_INSTALL_PREFIX=/install/dir /path/to/superbuild/git/repo
```
`/path/to/ome-model`

rleigh-codelibre replied:
Where did you see this?  It's definitely ome-model-dir on my `ome-model` branch.

rleigh-codelibre replied:
https://github.com/rleigh-codelibre/ome-cmake-superbuild/blob/ome-model/docs/sphinx/building.rst

sbesson replied:
Sorry, I meant that `bioformats` should become `ome-model`

rleigh-codelibre replied:
Ah, reading fail on my part.  Fixed now, and I also updated the README which was a bit stale.

sbesson commented on docs/sphinx/building.rst:
```diff
@@ -514,6 +514,6 @@ clone has the branch you wish to build checked out.  Run::
     mkdir build
     cd build
-    cmake -G Ninja -Dome-common-dir=/path/to/ome-common-cpp -Dome-xml-dir=/path/to/bioformats -Dome-files-dir=/path/to/ome-files -Dome-qtwidgets-dir=/path/to/ome-qtwidgets -Dbuild-packages=ome-qtwidgets -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/install/dir /path/to/superbuild/sources
+    cmake -G Ninja -Dome-common-dir=/path/to/ome-common-cpp -Dome-model-dir=/path/to/bioformats -Dome-files-dir=/path/to/ome-files -Dome-qtwidgets-dir=/path/to/ome-qtwidgets -Dbuild-packages=ome-qtwidgets -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/install/dir /path/to/superbuild/sources
```
`/path/to/ome-model`

sbesson reviewed (commented):
With this change, the OME Files C++ build using the new ome-model repo as shown by https://ci.openmicroscopy.org/job/OME-FILES-CPP-DEV-merge-trigger/.  A couple of inline comments but otherwise looks functional.

Three additional questions:
- as we switch all OME Files development branches to `master` should we do the same on `ome-cmake-superbuild` (and test the new GH functionality to change the base branch) ?
- the Model API documenttion is missing from the generated Sphinx documentation but I assume this requires some minimal work on ome-model
- do I undertand we are  renaming the `ome-xml` C++ component as `ome-model` ?  If are there implications for the package naming?

rleigh-codelibre commented:
"as we switch all OME Files development branches to master should we do the same on ome-cmake-superbuild (and test the new GH functionality to change the base branch) ?"

I think it would make sense for consistency; ome-common-cpp also needs updating?  I switched over ome-files-cpp and ome-qtwidgets after the branches got created.

"the Model API documenttion is missing from the generated Sphinx documentation but I assume this requires some minimal work on ome-model"

I think this is due to the points you mentioned.  I'll fix these up shortly.

"do I undertand we are renaming the ome-xml C++ component as ome-model ? If are there implications for the package naming?"

I've renamed the package inside the superbuild to match the git repo it's building.  The library name built inside the package is unchanged (still `ome-xml`).  This matches the java behaviour: the "ome-model" repository builds an ome-xml jar/lib for both languages, kept that way for compatibility.  We can rename it, but that will be a compatibility break, so maybe best left for 6.0?

sbesson commented:
Switching to `master` across the board would certainly make sense. :+1: for keeping the library name as `ome-xml` for now. Reading naively through the Sphinx documentation, figuring out the mapping between the prerequisite/repository name and the library name/package namespace is not trivial though. As a follow-up of this PR, this might something we could clarify.

rleigh-codelibre commented:
@sbesson If we go ahead with moving the model docs into ome-model, we can properly document things there.

hflynn commented:
I think there's plenty of scope to extend the developer section of the model docs when we migrate them, yes.

Commit:
docs: Drop reference to bioformats

Commit:
Update README

rleigh-codelibre commented:
Updated docs contain the last tweak: http://www.openmicroscopy.org/site/support/ome-files-cpp-staging/ome-cmake-superbuild/manual/html/building.html

@sbesson Please could this be merged?  I have some PRs to open which depend upon this.

sbesson commented:
I assume the latest changes are being currently run as of https://ci.openmicroscopy.org/view/Failing/job/OME-FILES-CPP-DEV-merge-trigger/381/ ? Could you relist this PR for review this afternoon?

rleigh-codelibre commented:
Will do.

sbesson commented:
Last commits look good to merge to me once the jobs turn green. Should I update all default branches to use `master` for the various `cpp` repos as well?

rleigh-codelibre commented:
@sbesson Yes, please do change the defaults to `master`.  Today's merge job is using master for every C++ repository.

rleigh-codelibre commented:
@sbesson All the CI jobs are green now.  Please could we merge it?
